edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentController.cs

```diff
@@ -0,0 +1,372 @@
+                .Select(
+                    service =>
+                    {
+                        Events.UpdateService(service);
+                        return this.client.ReplaceNamespacedServiceAsync(service, service.Metadata.Name, this.deviceNamespace);
+                    });
+            await Task.WhenAll(updatingTask);
+
+            // Delete all existing services that are not in desired list
+            var removingTasks = diff.Removed
+                .Select(
+                    name =>
+                    {
+                        Events.DeleteService(name);
+                        return this.client.DeleteNamespacedServiceAsync(name, this.deviceNamespace);
+                    });
+            await Task.WhenAll(removingTasks);
+
+            // Create new desired services
+            var addingTasks = diff.Added
+                .Select(
+                    service =>
+                    {
+                        Events.CreateService(service);
+                        return this.client.CreateNamespacedServiceAsync(service, this.deviceNamespace);
+                    });
+            await Task.WhenAll(addingTasks);
+        }
+
+        static Dictionary<string, string> GetServiceConfig(V1ServiceList services)
+        {
+            var config = new Dictionary<string, string>();
+            foreach (var service in services.Items)
+            {
+                if (service?.Metadata?.Name == null)
+                {
+                    Events.InvalidCreationString(service == null ? "service" : "name", "null service in list");
+                    continue;
+                }
+
+                if (service.Metadata?.Annotations == null || !service.Metadata.Annotations.TryGetValue(KubernetesConstants.CreationString, out string creationString))
+                {
+                    Events.InvalidCreationString(service.Kind, service.Metadata?.Name);
+
+                    var serviceWithoutStatus = new V1Service(service.ApiVersion, service.Kind, service.Metadata, service.Spec);
+                    creationString = JsonConvert.SerializeObject(serviceWithoutStatus);
+                }
+
+                config[service.Metadata.Name] = creationString;
+            }
+
+            return config;
+        }
+
+        async Task ManageDeployments(V1DeploymentList existing, IEnumerable<V1Deployment> desired)
+        {
+            // Compose creation strings from existing items
+            Dictionary<string, string> creationStrings = GetDeploymentConfig(existing);
+
+            // Update only deployments if configurations have not matched or it's an EdgeAgent with the same version
+            bool ShouldUpdate(V1Deployment deployment)
+            {
+                // current module is to update an existing one
+                if (deployment.Metadata.Annotations[KubernetesConstants.CreationString] != creationStrings[deployment.Metadata.Name])
```

Comment:
This logic probably needs to move lower in this function.

Reply:
Got rid of this method